src/wallet/rpcwallet.cpp

```diff
@@ -760,6 +764,171 @@ UniValue sc_create(const UniValue& params, bool fHelp)
+            "\nCreate a Side chain.\n"
+            "\nArguments:\n"
+            "{\n"                     
+            "   \"scid\": id                      (string, optional) The uint256 side chain ID, if omitted a random value is generated\n"
+            "   \"withdrawalEpochLength\": epoch  (numeric, optional, default=100) length of the withdrawal epochs\n"
+            "   \"fromaddress\":taddr             (string, optional) The taddr to send the funds from. If omitted funds are taken from all available UTXO\n"
+            "   \"toaddress\":scaddr              (string, required) The receiver PublicKey25519Proposition in the SC\n"
+            "   \"amount\":amount                 (numeric, required) Value expressed in " + CURRENCY_UNIT + "\n"
+            "   \"minconf\":conf                  (numeric, optional, default=1) Only use funds confirmed at least this many times.\n"
+            "   \"fee\":fee                       (numeric, optional, default="
+            + strprintf("%s", FormatMoney(SC_RPC_OPERATION_DEFAULT_MINERS_FEE)) + ") The fee amount to attach to this transaction.\n"
+            "}\n"
+            "\nResult:\n"
+            "\"transactionid\"    (string) The resulting transaction id.\n"
+            "\nExamples:\n"
+            + HelpExampleCli("create_sidechain", "'{\"toaddress\": \"8aaddc9671dc5c8d33a3494df262883411935f4f54002fe283745fb394be508a\" ,\"amount\": 5.0}]'")
+        );
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    // valid input keywords
+    static const std::set<std::string> validKeyArgs =
+        {"scid", "withdrawalEpochLength", "fromaddress", "toaddress", "amount", "minconf", "fee"};
+
+    UniValue inputObject = params[0].get_obj();
+
+    if (!inputObject.isObject())
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, expected object");
+
+    // keywords set in cmd
+    std::set<std::string> setKeyArgs;
+
+    // sanity check, report error if unknown/duplicate key-value pairs
+    for (const string& s : inputObject.getKeys())
+    {
+        if (!validKeyArgs.count(s))
+            throw JSONRPCError(RPC_INVALID_PARAMETER, string("Invalid parameter, unknown key: ") + s);
+
+        if (setKeyArgs.count(s))
+            throw JSONRPCError(RPC_INVALID_PARAMETER, string("Duplicate key in input: ") + s);
+
+        setKeyArgs.insert(s);
+    }
+
+    // ---------------------------------------------------------
+    uint256 scId;
+    if (setKeyArgs.count("scid"))
+    {
+        string inputString = find_value(inputObject, "scid").get_str();
+        if (inputString.length() == 0 || inputString.find_first_not_of("0123456789abcdefABCDEF", 0) != std::string::npos)
+            throw JSONRPCError(RPC_TYPE_ERROR, "Invalid scid format: not an hex");
+        scId.SetHex(inputString);
+    }
+    else
+    {
+        scId = GetRandHash();
```

Comment:
If I understand right there is no guarantee here that random scId is not taken already.
It should not happen in gazillion times and user can always retry, but maybe it worth making her succeeding on her first attempt? By doing something like
do { scId = GetRandHash()  } while ( !ScMgr::instance().sidechainExists(scId) );

Reply:
A clash is highly unlikely but theoretically might happen. In that case the command will fail anyway because the clash will be handled when the generated tx is checked.